cluster-up/check.sh

```diff
@@ -18,34 +18,46 @@
 #
 
 set -e
+fail=''
 if [ ! -c /dev/kvm ]; then
-	echo "[ERR ] missing /dev/kvm"
+    echo "[ERR ] missing /dev/kvm"
+    fail=1
 else
-	echo "[ OK ] found /dev/kvm"
+    echo "[ OK ] found /dev/kvm"
 fi
 
 KVM_ARCH=""
 KVM_NESTED="unknown"
 if [ -f "/sys/module/kvm_intel/parameters/nested" ]; then
-	KVM_NESTED=$( cat /sys/module/kvm_intel/parameters/nested )
-	KVM_ARCH="intel"
+    KVM_NESTED=$(cat /sys/module/kvm_intel/parameters/nested)
+    KVM_ARCH="intel"
 elif [ -f "/sys/module/kvm_amd/parameters/nested" ]; then
-	KVM_NESTED=$( cat /sys/module/kvm_amd/parameters/nested )
-	KVM_ARCH="amd"
+    KVM_NESTED=$(cat /sys/module/kvm_amd/parameters/nested)
+    KVM_ARCH="amd"
 fi
 
 function is_enabled() {
-	if [ "$1" == "1" ]; then
-		return 0
-	fi
-	if [ "$1" == "Y" ] || [ "$1" == "y"]; then
-		return 0
-	fi
-	return 1
+    if [ "$1" == "1" ]; then
+        return 0
+    fi
+    if [ "$1" == "Y" ] || [ "$1" == "y" ]; then
+        return 0
+    fi
+    return 1
 }
 
 if is_enabled "$KVM_NESTED"; then
-	echo "[ OK ] $KVM_ARCH nested virtualization enabled"
+    echo "[ OK ] $KVM_ARCH nested virtualization enabled"
 else
-	echo "[ERR ] $KVM_ARCH nested virtualization not enabled"
+    echo "[ERR ] $KVM_ARCH nested virtualization not enabled"
+    fail=1
+fi
+
+if [[ "$KUBEVIRT_PROVIDER" =~ external|local ]]; then
+    echo "nested virtualization not required"
+elif [ -z "$fail" ]; then
```

Comment:
do we have a non external | local that use Emulation (kubevirt configmap has it IIRC ?) so it doesn't need kvm ?

Reply:
Sorry, I don't understand your question?

Reply:
https://github.com/kubevirt/kubevirt/blob/ed8da106d0eb177b0d468a6b8db4a9b6de38115f/docs/software-emulation.md
in case useEmulation is enabled, `/dev/kvm` is allowed to not exists even on the non `external|local` providers ?